core/src/main/java/org/elasticsearch/action/admin/indices/shrink/ShrinkRequest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.action.admin.indices.shrink;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.admin.indices.create.CreateIndexRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+/**
+ * Request class to shrink an index into a single shard
+ */
+public class ShrinkRequest extends AcknowledgedRequest<ShrinkRequest> implements IndicesRequest {
+
+    private CreateIndexRequest skrinkIndexRequest;
+    private String sourceIndex;
+
+    ShrinkRequest() {}
+
+    public ShrinkRequest(String targetIndex, String sourceindex) {
+        this.skrinkIndexRequest = new CreateIndexRequest(targetIndex);
+        this.sourceIndex = sourceindex;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
```

Comment:
since we expose the underlying createIndexRequest, we should validate it too. Also we can do some early validation here like the number of shards and having no mapping etc. Last, I *wonder* if things will be simpler conceptually if ShrinkRequest will extend CreateIndexRequest and have all it's properties directly accessible (and adding one "SourceIndex" property)